### PR TITLE
Add run-tests action

### PR DESCRIPTION
**Pull request by jonashaag**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jonashaag** [comment]
@mroeschke

File: .github/workflows/python-dev.yml [hunk 1]
```diff
@@ -72,25 +72,5 @@ jobs:
...
-      run: |
-        coverage report -m
-
-    - name: Upload coverage to Codecov
-      uses: codecov/codecov-action@v2
-      with:
-        flags: unittests
-        name: codecov-pandas
-        fail_ci_if_error: true
+    - name: Test
```
Comment (mroeschke) [inline .github/workflows/python-dev.yml:75]: Could you link to the build where this was run (currently this is disabled)?

Comment (mroeschke) [reply .github/workflows/python-dev.yml:75]: (Not in the yaml, but here in the comments)

Comment (jonashaag) [reply .github/workflows/python-dev.yml:76]: Can't find it, will re-run. Using NumPy stable because nightly was broken the other day.

Comment (jonashaag) [reply .github/workflows/python-dev.yml:76]: There seem to be errors that I'm not sure are related to this PR. Could you have a look?

Also I realised that in the dev build we have `fail_ci_if_error: true` (https://github.com/pandas-dev/pandas/blob/44b660dc4a07f4fb507c31795ae63dca2e6e9440/.github/workflows/python-dev.yml#L96) but everywhere else we have `fail_ci_if_error: false`. Shall I make this a flag of the `run-tests` action?

Comment (mroeschke) [reply .github/workflows/python-dev.yml:76]: Thanks looks like the tests are running with expected failures.

Also yes, `fail_ci_if_error: false` should be set in the `run-tests` action.

**Review by mroeschke — COMMENTED** [review]
Looks good after reverting the changes to `.github/workflows/python-dev.yml` used for testing.

File: .github/workflows/python-dev.yml (hunk 1 above)
Comment (jonashaag) [reply .github/workflows/python-dev.yml:76]: Set to `false` for all invocations or keep current behaviour?

Comment (mroeschke) [reply .github/workflows/python-dev.yml:76]: `fail_ci_if_error: false` should be used for all `codecov/codecov-action@v2` actions invocations

File: .github/workflows/python-dev.yml [hunk 2]
```diff
@@ -57,40 +57,20 @@ jobs:
...
         pip install git+https://github.com/nedbat/coveragepy.git
-        pip install cython python-dateutil pytz hypothesis pytest>=6.2.5 pytest-xdist pytest-cov
+        pip install cython python-dateutil pytz hypothesis pytest>=6.2.5 pytest-xdist pytest-cov pytest-asyncio>=0.17
         pip list
 
     - name: Build Pandas
       run: |
         python setup.py build_ext -q -j2
-        python -m pip install -e . --no-build-isolation --no-use-pep517
+        pip install -e . --no-build-isolation --no-use-pep517
```
Comment (mroeschke) [inline .github/workflows/python-dev.yml:69]: Same comment here about using `python 3 -m ...`

File: .github/workflows/python-dev.yml [hunk 3]
```diff
@@ -57,40 +57,20 @@ jobs:
     - name: Install dependencies
       shell: bash -el {0}
       run: |
-        python -m pip install --upgrade pip setuptools wheel
```
[the remark this answers is not in the document]
Comment (jonashaag) [reply .github/workflows/python-dev.yml:60]: OK, looks good with `python3`.

Comment (mroeschke) [inline .github/workflows/python-dev.yml:60]: Was `python3 -m pip install --upgrade pip setuptools wheel` intentionally removed?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Awesome, thanks @jonashaag

**Pull request merged**